compiler/qsc_partial_eval/tests/operators.rs

```diff
@@ -300,3 +303,2065 @@ fn logical_not_unary_operator_generates_logical_not_rir_instruction() {
+                a * b
+            }
+        }
+        "#,
+    });
+    let measurement_callable_id = CallableId(1);
+    assert_callable(
+        &program,
+        measurement_callable_id,
+        &expect![[r#"
+            Callable:
+                name: __quantum__qis__mresetz__body
+                call_type: Measurement
+                input_type:
+                    [0]: Qubit
+                    [1]: Result
+                output_type: <VOID>
+                body: <NONE>"#]],
+    );
+    let readout_callable_id = CallableId(2);
+    assert_callable(
+        &program,
+        readout_callable_id,
+        &expect![[r#"
+        Callable:
+            name: __quantum__qis__read_result__body
+            call_type: Readout
+            input_type:
+                [0]: Result
+            output_type: Boolean
+            body: <NONE>"#]],
+    );
+    let output_record_id = CallableId(3);
+    assert_callable(
+        &program,
+        output_record_id,
+        &expect![[r#"
+            Callable:
+                name: __quantum__rt__integer_record_output
+                call_type: OutputRecording
+                input_type:
+                    [0]: Integer
+                    [1]: Pointer
+                output_type: <VOID>
+                body: <NONE>"#]],
+    );
+    assert_blocks(
+        &program,
+        &expect![[r#"
+            Blocks:
+            Block 0:Block:
+                Call id(1), args( Qubit(0), Result(0), )
+                Variable(0, Boolean) = Call id(2), args( Result(0), )
+                Variable(1, Boolean) = Icmp Eq, Variable(0, Boolean), Bool(false)
+                Branch Variable(1, Boolean), 2, 3
+            Block 1:Block:
+                Call id(1), args( Qubit(0), Result(1), )
+                Variable(3, Boolean) = Call id(2), args( Result(1), )
+                Variable(4, Boolean) = Icmp Eq, Variable(3, Boolean), Bool(false)
+                Branch Variable(4, Boolean), 5, 6
+            Block 2:Block:
+                Variable(2, Integer) = Store Integer(0)
+                Jump(1)
+            Block 3:Block:
+                Variable(2, Integer) = Store Integer(1)
+                Jump(1)
+            Block 4:Block:
+                Variable(6, Integer) = Mul Variable(2, Integer), Variable(5, Integer)
+                Call id(3), args( Variable(6, Integer), Pointer, )
+                Return
+            Block 5:Block:
+                Variable(5, Integer) = Store Integer(1)
+                Jump(4)
+            Block 6:Block:
+                Variable(5, Integer) = Store Integer(0)
+                Jump(4)"#]],
+    );
+}
+
+#[test]
+fn integer_div_with_lhs_classical_integer_and_rhs_dynamic_integer() {
+    let program = get_rir_program(indoc! {
+        r#"
+        namespace Test {
+            @EntryPoint()
+            operation Main() : Int {
+                use q = Qubit();
+                let i = MResetZ(q) == Zero ? 0 | 1;
+                1 / i
```

Comment:
Could you include a test case that will trigger the division by zero error? Something like this:
```qsharp
use q = Qubit();
let z = 0;
let i = MResetZ(q) == Zero ? 0 | 1;
i / z;
```

Reply:
Added test case for division by zero.